src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Changed.cs

```diff
@@ -11,15 +12,18 @@ public class Directory_Changed_Tests : FileSystemWatcherTest
         [Fact]
         public void FileSystemWatcher_Directory_Changed_LastWrite()
         {
-            using (var testDirectory = new TempDirectory(GetTestFilePath()))
-            using (var dir = new TempDirectory(Path.Combine(testDirectory.Path, "dir")))
-            using (var watcher = new FileSystemWatcher(testDirectory.Path, Path.GetFileName(dir.Path)))
+            RetryHelper.Execute(() =>
             {
-                Action action = () => Directory.SetLastWriteTime(dir.Path, DateTime.Now + TimeSpan.FromSeconds(10));
+                using (var testDirectory = new TempDirectory(GetTestFilePath()))
```

Comment:
I wonder why is this temp directory being creates inside the test file path instead of using the default constructor, which creates the folder in the system default temp folder.

Reply:
as above